runtime/doc/msgpack_rpc.txt

```diff
@@ -252,4 +253,169 @@ RPC functions are available in Vimscript:
 a nvim server. |v:servername|
 
 ==============================================================================
+7. Remote UIs					           *rpc-remote-ui*
+
+Nvim allows Graphical user interfaces to be implemented by separate processes
+communicating with Nvim over the RPC api. Currently the ui model conists of a
+terminal-like grid with one single, monospace font size, with a few elements
+that could be drawn separately from the grid (for the momemnt only the popup
+menu)
+
+After connecting to a nvim instance (typically a spawned, embedded instance)
+use the |nvim_ui_attach|(width, height, options) api method to tell nvim that your
+program wants to draw the nvim screen on a grid with "width" times
+"height" cells. "options" should be a dictionary with the following (all
+optional) keys:
+	`rgb`:			recieve rgb color values with 8-bit per channels
+				if true, terminal color palette numbers if
+				false. Defaults to true.
+	`popupmenu_external`:	Instead of drawing the completion popupmenu on
+				the grid, Nvim will send higher-level events to
+				the ui and let it draw the popupmenu.
+				Defaults to false.
+
+Nvim will then send msgpack-rpc notifications, with the event name "redraw"
+and a single argument, which is an array of screen updates, described below,
+that should be processed in order. Preferably, the user should only be able to
+see the screen state after all updates are processed, not any intermediate
+state after processing only a part of the array.
+
+Screen updates are arrays, where the first element a string decribing the kind
+of update:
+
+["resize", width, height]
+	The internal screen is resized to `width` and `height` cells.
+
+["clear"]
+	Clear the screen. The screen is filled with the default
+	background color.
```

Comment:
Just setting the background color wouldn't fill the screen, no? But you're right "Clear the screen" is enough here.

> There is also the fact that the events are mostly written with the TUI interface in mind, but that's another problem.

This is because is the vim/nvim internal model pretty much is a terminal grid currently (with a few additions as I mentioned)